datafusion/core/tests/repartition_exec_blocks.rs

```diff
@@ -0,0 +1,60 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use arrow::array::UInt32Array;
+use arrow::datatypes::{DataType, Field, Schema};
+use arrow::record_batch::RecordBatch;
+use datafusion::physical_plan::coalesce_partitions::CoalescePartitionsExec;
+use datafusion::physical_plan::repartition::RepartitionExec;
+use datafusion::physical_plan::{ExecutionPlan, Partitioning};
+use datafusion::prelude::{SessionConfig, SessionContext};
+use datafusion::test_util::UnboundedExec;
+use datafusion_common::from_slice::FromSlice;
+use datafusion_common::Result;
+use datafusion_physical_expr::expressions::Column;
+use datafusion_physical_expr::PhysicalExpr;
+use futures::StreamExt;
+use std::sync::Arc;
+
+/// See <https://github.com/apache/arrow-datafusion/issues/5278>
+#[tokio::test]
+async fn unbounded_repartition_sa() -> Result<()> {
```

Comment:
I think we can insert a timeout here in case of an error. You can check `unbounded_file_with_swapped_join` for an example.

Reply:
I cannot get that timeout macro to work reliably. It only seems to work w/ multi-thread tokio executors that have more than 1 thread. However than the test itself is a bit of a lucky shot either, because the main test method will no longer block just because the repartition task spins forever. So I guess we have to live w/o a timeout.